modules/gltf/editor/editor_import_blend_runner.cpp

```diff
@@ -268,7 +276,22 @@ Error EditorImportBlendRunner::do_import_rpc(const Dictionary &p_options) {
 			}
 			case HTTPClient::STATUS_BODY: {
 				client->poll();
-				// Parse response here if needed. For now we can just ignore it.
+				PackedByteArray response_chunk = client->read_response_body_chunk();
```

Comment:
Following your comments, I used an easier method to determine if the exportation is successful or not. I just look if the string 'BLENDER_GODOT_EXPORT_SUCCESSFUL' is present in the output. That way, I don't have to parse the XML and the success message should always be in the first chunk. Worst case scenario, the error message is incomplete which, I think it's not a big deal (we did not have any error message before).

Reply:
If you're introducing a string to look for, I think it would be safer to read the whole buffer. It's not too much extra work.

Something like

```
	// Wait for response.
	bool done = false;
	PackedByteArray response;
	while (!done) {
		status = client->get_status();
		switch (status) {
			case HTTPClient::STATUS_REQUESTING: {
				client->poll();
				break;
			}
			case HTTPClient::STATUS_BODY: {
				client->poll();
				response.append_array(client->read_response_body_chunk());
				done = true;
				break;
			}
			default: {
				ERR_FAIL_V_MSG(ERR_CONNECTION_ERROR, vformat("Unexpected status during RPC response: %d", status));
			}
		}
	}

	String response_text = "No response from Blender.";
	if (response.size() > 0) {
		response_text = String::utf8((const char *)response_chunk.ptr(), response_chunk.size());
	}

	if (client->get_response_code() != HTTPClient::RESPONSE_OK) {
		ERR_FAIL_V_MSG(ERR_QUERY_FAILED, vformat("Error received from Blender - status code: %s, error: %s", client->get_response_code(), response_text));
	} else if (response_text.find("BLENDER_GODOT_EXPORT_SUCCESSFUL") < 0) {
		// Previous versions of Godot used a Python script where the RPC function did not return
		// a value, causing the error 'cannot marshal None unless allow_none is enabled'.
		// If an older version of Godot is running and has started Blender with this script,
		// we will receive the error, but there's a good chance that the import was successful.
		// We are discarding this error to maintain backward compatibility and prevent situations
		// where the user needs to close the older version of Godot or kill Blender.
		if (response_text.find("cannot marshal None unless allow_none is enabled") < 0) {
			ERR_FAIL_V_MSG(ERR_QUERY_FAILED, vformat("Blender exportation failed: %s", response_text));
		}
	}

	return OK;
}

```

Reply:
Actually that's wrong, let me check what's the right way :)

Reply:
Updated the comment above

Reply:
Thanks a lot for this code. I never worked with Godot HTTPClient before. It seems to work perfectly.